src/Breadcrumbs.php

```diff
@@ -147,77 +180,159 @@ public function links(array $value): self
+    {
+        $listAttributes = $this->listAttributes;
+        $classes = $listAttributes['class'] ?? null;
+
+        $listId = match ($this->listId) {
+            true => $listAttributes['id'] ?? Html::generateId(self::LIST_NAME . '-'),
+            '', false => null,
+            default => $this->listId,
+        };
+
+        unset($listAttributes['class'], $listAttributes['id']);
+
+        $items = [];
+
+        foreach ($this->links as $link) {
+            $items[] = $this->renderItem($link);
+        }
+
+        $items = implode("\n", $items);
+
+        if ($items === '') {
+            return '';
+        }
+
+        Html::addCssClass($listAttributes, [self::LIST_NAME, $classes]);
+
+        if ($this->listTagName === '') {
+            throw new InvalidArgumentException('List tag cannot be empty.');
+        }
+
+        return Html::tag($this->listTagName)
+            ->attributes($listAttributes)
+            ->content("\n", $items, "\n")
+            ->id($listId)
+            ->encode(false)
+            ->render();
+    }
+
     /**
      * Renders a single breadcrumb item.
      *
-     * @param array $link the link to be rendered. It must contain the "label" element. The "url" element is optional.
-     * @param string $template the template to be used to rendered the link. The token "{link}" will be replaced by the
-     * link.
-     *
-     * @throws JsonException|RuntimeException if `$link` does not have "label" element.
+     * @param Link $link The breadcrumb item to render.
      *
-     * @return string the rendering result
+     * @return string The rendering result.
      */
-    private function renderItem(array $link, string $template): string
+    private function renderItem(Link $link): string
     {
-        $encodeLabel = ArrayHelper::remove($link, 'encode', $this->encodeLabels);
-
-        if (array_key_exists('label', $link)) {
-            $label = $encodeLabel ? Html::encode($link['label']) : $link['label'];
-        } else {
+        if ($link->label === '') {
             throw new RuntimeException('The "label" element is required for each link.');
         }
 
-        if (isset($link['template'])) {
-            $template = $link['template'];
-        }
+        $itemsAttributes = $this->itemAttributes;
+        $classes = $itemsAttributes['class'] ?? null;
+
+        unset($itemsAttributes['class']);
+
+        $linkTag = $this->renderLink($link);
+        Html::addCssClass($itemsAttributes, [self::ITEM_NAME, $classes]);
 
-        if (isset($link['url'])) {
-            $options = $link;
-            unset($options['template'], $options['label'], $options['url']);
-            $linkHtml = Html::a($label, $link['url'], $options)->encode($this->encodeTags);
-        } else {
-            $linkHtml = $label;
+        if ($link->url === null) {
+            $itemsAttributes['aria-current'] = 'page';
+
+            Html::addCssClass($itemsAttributes, $this->itemActiveClass);
```

Comment:
Active item can be with url. Suggest add property "bool $active = false" to `Link` class and check it.

Reply:
I'm going to add it but the documentation is clear, the last element does not have a url.

Reply:
done.